core/src/avm2/object.rs

```diff
@@ -413,10 +454,18 @@ pub trait TObject<'gc>: 'gc + Collect + Debug + Into<Object<'gc>> + Clone + Copy
         multiname: &Multiname<'gc>,
         value: Value<'gc>,
         activation: &mut Activation<'_, 'gc>,
+        ic: Option<&mut InlineCache<Property>>,
     ) -> Result<(), Error<'gc>> {
-        match self.vtable().get_trait(multiname) {
-            Some(Property::Slot { slot_id }) => {
-                let base = self.base();
+        // Try ic
+        let base = self.base();
+        if let Some(shape_id) = base.shape_id() {
+            if self.try_set_property_by_shape(activation, multiname, value, shape_id, ic)? {
+                return Ok(());
+            }
+        }
+
+        match self.vtable().get_trait_with_ns(multiname) {
```

Comment:
i don't think we need to use `get_trait_with-ns`, even the original solution is not using it